JWT/JWT.cs

```diff
@@ -18,17 +20,31 @@ public enum JwtHashAlgorithm
     /// </summary>
     public static class JsonWebToken
     {
-        private static Dictionary<JwtHashAlgorithm, Func<byte[], byte[], byte[]>> HashAlgorithms;
+        private static readonly IDictionary<JwtHashAlgorithm, Func<byte[], string, byte[], byte[]>> HashAlgorithms;
         private static JavaScriptSerializer jsonSerializer = new JavaScriptSerializer();
 
         static JsonWebToken()
         {
-            HashAlgorithms = new Dictionary<JwtHashAlgorithm, Func<byte[], byte[], byte[]>>
+            HashAlgorithms = new Dictionary<JwtHashAlgorithm, Func<byte[], string, byte[], byte[]>>
+            {
+                { JwtHashAlgorithm.HS256, (key, _, value) => { using (var sha = new HMACSHA256(key)) { return sha.ComputeHash(value); } } },
+                { JwtHashAlgorithm.HS384, (key, _, value) => { using (var sha = new HMACSHA384(key)) { return sha.ComputeHash(value); } } },
+                { JwtHashAlgorithm.HS512, (key, _, value) => { using (var sha = new HMACSHA512(key)) { return sha.ComputeHash(value); } } },
+                { JwtHashAlgorithm.RS256, (key, secret, value) => RS256(key, secret, value) }
+            };
+        }
+
+        private static byte[] RS256(byte[] key, string secret, byte[] value)
+        {
+            var cert = new X509Certificate2(key, secret);
```

Comment:
Private key is a byte array in base64 representation (usually kept in Web.config for server-side web apps, you know). See [Google](https://developers.google.com/identity/protocols/OAuth2ServiceAccount) or [Auth0](https://auth0.com/docs/apiv2) for examples.

Secret (or password in term of MSDN) is just a string, see [X509Certificate2 ctor](https://msdn.microsoft.com/en-us/library/ms148418%28v=vs.110%29.aspx). For instance, for Google Service Account it's "notasecret", for client-side apps or at Auth0 it's something meaningful.

I think we can have an overload accepting either X509Certificate2, or byte[] and string and construct a cert by ourselves.
But such method will be usable only with RS256 and not with other algorithms.
On other hand, existing overload accepting secret is only usable with RS425 as well.